### PR TITLE
[BD] Upgrade test packages

### PR DESCRIPTION
**Pull request by morenol**
Upgrade test packages and some pylint fixes.

Remove mock usage in favor of unittest.mock

## Reviewers
- [ ] @jmbowman 
- [x] Ready for edX review.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @morenol! I've created [OSPR-4356](https://openedx.atlassian.net/browse/OSPR-4356) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: analytics_data_api/v0/tests/views/test_courses.py [hunk 1]
```diff
@@ -6,15 +6,15 @@
 import datetime
 from collections import OrderedDict
 from itertools import groupby
+from unittest.mock import Mock, patch
 
 import ddt
 import pytz
 import six
-import six.moves.urllib.parse  # pylint: disable=import-error
+import six.moves.urllib.parse  # pylint: disable=import-error,wrong-import-order
```
Comment (jmbowman) [inline analytics_data_api/v0/tests/views/test_courses.py:14]: You may be able to drop this line completely, it seems that `import six` above should already cover this.

File: analytics_data_api/v0/tests/views/test_programs.py [hunk 2]
```diff
@@ -31,37 +31,41 @@ def setUp(self):
...
+        course_ids = kwargs.pop('course_ids', None)
         if course_ids is None:
             course_ids = [[self.course_id]] * len(ids)
 
         return [self.expected_result(item_id, course_ids=course_id, **kwargs)
                 for item_id, course_id in zip(ids, course_ids)]
 
     # pylint: disable=arguments-differ
-    def expected_result(self, item_id, course_ids=None):
+    def expected_result(self, item_id, **kwargs):
```
Comment (jmbowman) [inline analytics_data_api/v0/tests/views/test_programs.py:66]: Is the `arguments-differ` warning disable still needed after this change?

File: requirements/test.in [hunk 3]
```diff
@@ -2,15 +2,14 @@
...
-coverage==4.4.1
-ddt==1.1.1
-diff-cover >= 0.9.9
+coverage
+ddt
+diff-cover
 django-dynamic-fixture
-mock==2.0.0
-pep257==0.7.0
+pep257
```
Comment (jmbowman) [inline requirements/test.in:9]: Newer versions of this are being released as the `pydocstyle` package, if you want to try upgrading it even further.

Comment (morenol) [reply requirements/test.in]: Done

**Comment by edx-webhook** [comment]
@morenol 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-analytics-data-api&entry.867055334=https://github.com/edx/edx-analytics-data-api/pull/362&entry.1484655318=https://github.com/morenol&entry.752974735=2020-04-03+15:48&entry.1917517419=2020-04-13+17:59&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**